pkg/controllers/nodepool/hash/controller.go

```diff
@@ -69,3 +78,32 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) operatorcontr
 		WithOptions(controller.Options{MaxConcurrentReconciles: 10}),
 	)
 }
+
+func (c *Controller) updateNodeClaimHash(ctx context.Context, np *v1beta1.NodePool) error {
+	ncList := &v1beta1.NodeClaimList{}
+	if err := c.kubeClient.List(ctx, ncList); err != nil {
+		return err
+	}
+
+	for i := range ncList.Items {
+		stored := ncList.Items[i].DeepCopy()
+		if ncList.Items[i].Labels[v1beta1.NodePoolLabelKey] != np.Name {
+			continue
+		}
+
+		if ncList.Items[i].Annotations[v1beta1.NodePoolHashVersionAnnotationKey] != v1beta1.NodePoolHashVersion && ncList.Items[i].StatusConditions().GetCondition(v1beta1.Drifted) == nil {
```

Comment:
Can we comment over this? Otherwise it's going to be a little unclear why we decide to do this check on the drifted status condition